streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java

```diff
@@ -0,0 +1,104 @@
+package org.ray.streaming.jobgraph;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Job graph, the logical plan of streaming job.
+ */
+public class JobGraph implements Serializable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JobGraph.class);
+
+  private final String jobName;
+  private final Map<String, String> jobConfig;
+  private List<JobVertex> jobVertexList;
+  private List<JobEdge> jobEdgeList;
+  private String digraph;
+
+  public JobGraph(String jobName, Map<String, String> jobConfig) {
+    this.jobName = jobName;
+    this.jobConfig = jobConfig;
+    this.jobVertexList = new ArrayList<>();
+    this.jobEdgeList = new ArrayList<>();
+  }
+
+  /**
+   * Generate direct graph by current job graph.
+   * @return Digraph in string type.
+   *
+   * Notice:
+   * This is a temporarily implement in hard code.
+   * May use 'guru.nidi:graphviz-java' in the future if needed.
+   */
+  public String generateDigraph() {
```

Comment:
we‘d better explain more about what is digraph and why it is needed in javadoc for this method or this class

Reply:
Optimized.

Reply:
Looks good now.